src/gatsby-node.js

```diff
@@ -21,12 +21,22 @@ See docs here – https://github.com/fabe/gatsby-source-figma
 
   const { createNode } = actions;
   let files = [];
+  let images = [];
   let project = {};
 
-  if (fileId) {
+  if (fileId && !nodeIds) {
+
     const file = await Figma.fetchFile(fileId, accessToken);
+    
     files = [file];
-  } else if (projectId) {
+  } else if (fileId && nodeIds) {
+    const imagesData = nodeIds.map(nodeId =>
+      Figma.fetchNode(fileId, nodeId, scale, format, accessToken)
+    );
+
+    images = await Promise.all(imagesData);
```

Comment:
If `fileId` _and_ `nodeIds` are set, both should be available for querying. So we still need to make the `Figma.fetchFile` request that is currently skipped when both are set. Maybe try putting the image querying into a different if statement.

Reply:
Updated!